root/etc/e-smith/templates/etc/nethserver/cockpit.otp.force/10base

```diff
@@ -1,10 +1,14 @@
+#
+# The user listed here are forced to use otp for cockpit
+#
+
 {
-  if ( -f '/root/.2fa.secret') {
+  if ( -e '/root/.2fa_cockpit.secret') {
       $OUT .= "root\n";
   }
   foreach my $name (glob('/var/lib/nethserver/home/*')) {
       $name =~ s|/var/lib/nethserver/home/||g;
-      next if ( ! -f "/var/lib/nethserver/home/$name/.2fa.secret");
+      next if ( ! -e "/var/lib/nethserver/home/$name/.2fa_cockpit.secret");
```

Comment:
```suggestion
      $secret = "/var/lib/nethserver/home/$name/.2fa_cockpit.secret";
      # check if link is valid
      next if ( lstat $secret and not stat $secret );
```

Reply:
it does not work with root @gsanchietti  ???? it is supposed to work with another user ?

Reply:
Just report the same check also above for user root.

Reply:
could you test the code, it seems it doesn't work like you expected

Reply:
Already tested, but feel free to re-test and change it.
The point is: you must not check only if the link exists, but also the link must not be broken.

Reply:
it seems the behaviour is different between a script and inside the template expansion, however I do not know why

Reply:
Your code seems to already correctly check for broken links.